python/ray/serve/_private/deployment_state.py

```diff
@@ -1506,24 +1520,40 @@ def autoscale(
             current_num_ongoing_requests=current_num_ongoing_requests,
             current_handle_queued_queries=current_handle_queued_queries,
         )
-        if decision_num_replicas == self._target_state.num_replicas:
-            return
 
-        # Adjust the logged replica count based on `target_capacity`. In reality the
-        # autoscaling algorithm operates on the full replica count and the target
-        # will be adjusted in the in the main deployment update cycle, but this log
-        # statement needs to be comprehensible to the user.
-        adjusted_decision_num_replicas = self.get_capacity_adjusted_num_replicas(
-            decision_num_replicas, target_capacity
+        # Clip the replica count by capacity-adjusted bounds.
+        # TODO (shrekris-anyscale): this should logic should be pushed into the
```

Comment:
@GeneDer We should consider how to handle `target_capacity` in future when we think about approaches for custom scaling.

Reply:
Good call, but in my mind `target_capacity` shouldn't affect custom scaling since we are trying to give users full control for how they want to scale. All current autoscaling configs will be ignored if the new custom scaling config is set🙂